### PR TITLE
Clarify and correct initialization content re: Docker

### LINKED ISSUE #2224: Clarify docker init instructions in Install Sensu doc
### Description
Comment in https://sensucommunity.slack.com/archives/C68LV5M9U/p1582803992027300 (also raised by other users) about Docker initialization. 

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [ ] update (add missing or refresh existing content)
- [x] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/latest/installation/install-sensu/#2-configure-and-start
https://docs.sensu.io/sensu-go/latest/installation/install-sensu/#3-initialize
https://docs.sensu.io/sensu-go/latest/reference/backend/#initialization
https://docs.sensu.io/sensu-go/latest/sensuctl/reference/#username-password-and-namespace
https://docs.sensu.io/sensu-go/latest/reference/rbac/#default-users

### Expected behavior
Install Sensu doc should have clear instructions and explain any differences between Docker installs and other platforms.

### Current behavior
Install Sensu doc is confusing for Docker users.

### Corrections needed
- [ ] Explain Docker initialization in 2. Configure and start
- [ ] Remove Docker from 3. Initialization (except for a note explaining exception for Docker)
- [ ] Clarify initialization information in backend reference

- Include the initialization with the "configure and start" instructions for Docker & Docker compose, and add a note that explains that initialization is handled differently in our Docker container.
- On a VM you start the backend first, then initialize; on Docker you set env vars first then initialization happens automatically via the entrypoint script. docker init happens automatically via the entry point, using the old defaults if env vars aren’t provided to override.

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Updates code examples and instructions for Docker vs. Ubuntu/Debian and RHEL/CentOS in:
     - https://docs.sensu.io/sensu-go/latest/installation/install-sensu/#2-configure-and-start
     - https://docs.sensu.io/sensu-go/latest/installation/install-sensu/#3-initialize
- Clarifies when initialization happens for Docker vs. Ubuntu/Debian and RHEL/CentOS in:
     - https://docs.sensu.io/sensu-go/latest/installation/install-sensu/#2-configure-and-start
     - https://docs.sensu.io/sensu-go/latest/installation/install-sensu/#3-initialize
     - https://docs.sensu.io/sensu-go/latest/reference/backend/#initialization
- Adds minor link changes and language clarifications in:
     - https://docs.sensu.io/sensu-go/latest/sensuctl/reference/#username-password-and-namespace
     - https://docs.sensu.io/sensu-go/latest/reference/rbac/#default-users

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2224

### CODE REVIEW

**Comment by hillaryfraley** [comment]
@cwjohnston I know this is a bit different than what we talked about, so if any of my decisions here don't make sense or you disagree with my approach, please let me know. I basically decided to present "set via env vars" as the way to go in the installation guide and save the default credentials explanation for the backend reference.

File: content/sensu-go/5.18/reference/backend.md [hunk 1]
```diff
@@ -51,8 +51,58 @@ For information about creating and managing checks, see:
 
 ## Initialization
 
-For a **new** installation, you must set up an administrator username and password.
-To do this, set environment variables as shown below, replacing `YOUR_USERNAME` and `YOUR_PASSWORD` with the username and password you want to use:
+For a **new** installation, you can set administrator credentials with environment variables to override the default admin username (`admin`) and password (`P@ssw0rd!`).
+
+- If you are using Docker, you can set admin credentials during installation step [2. Configure and start][24].
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/backend.md]:
```suggestion
- If you are using Docker, you can use environment variables to override the default admin username (`admin`) and password (`P@ssw0rd!`) during installation step [2. Configure and start][24].
```

File: content/sensu-go/5.18/reference/backend.md [hunk 2]
```diff
@@ -51,8 +51,58 @@ For information about creating and managing checks, see:
 
 ## Initialization
 
-For a **new** installation, you must set up an administrator username and password.
-To do this, set environment variables as shown below, replacing `YOUR_USERNAME` and `YOUR_PASSWORD` with the username and password you want to use:
+For a **new** installation, you can set administrator credentials with environment variables to override the default admin username (`admin`) and password (`P@ssw0rd!`).
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/backend.md]:
```suggestion
For a **new** installation, the backend database must be initialized by providing a username and password to be granted administrative privileges. Although initialization is required for every new installation, the implementation differs depending on your method of installation:
```

File: content/sensu-go/5.18/reference/backend.md [hunk 3]
```diff
@@ -51,8 +51,58 @@ For information about creating and managing checks, see:
...
+
+- If you are using Docker, you can set admin credentials during installation step [2. Configure and start][24].
+- If you are using Ubuntu/Debian or RHEL/CentOS, you can set admin credentials during installation step [3. Initialization][25].
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/backend.md]:
```suggestion
- If you are using Ubuntu/Debian or RHEL/CentOS, you must specify admin credentials during installation step [3. Initialization][25]. Sensu does not apply a default admin username or password for Ubuntu/Debian or RHEL/CentoOS installations.
```

File: content/sensu-go/5.18/reference/backend.md [hunk 4]
```diff
@@ -51,8 +51,58 @@ For information about creating and managing checks, see:
...
+
+This step bootstraps the first admin user account for your Sensu installation.
+This account will be granted the cluster admin role.
+
+If you do not include the environment variables to set administrator credentials, the backend will initialize with the default admin credentials (username `admin` and password `P@ssw0rd!`).
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/backend.md]:
```suggestion
```

File: content/sensu-go/5.18/sensuctl/reference.md [hunk 5]
```diff
@@ -62,14 +62,16 @@ The default URL is `http://127.0.0.1:8080`.
 To connect to a [Sensu cluster][7], connect sensuctl to any single backend in the cluster.
 For information about configuring the Sensu backend URL, see the [backend reference][5].
 
-### Username, Password, and Namespace
+### Username, password, and namespace
+
+During the [Sensu backend installation][40] process, you create an administrator username and password for a `default` namespace.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/sensuctl/reference.md]:
```suggestion
During the [Sensu backend installation][40] process, you create an administrator username and password and a `default` namespace.
```

File: content/sensu-go/5.18/reference/rbac.md [hunk 6]
```diff
@@ -242,7 +242,7 @@ Use your Sensu username and password to [configure sensuctl][26] or log in to th
 
 ### Default users
 
-When you install the Sensu backend, during the [initialization step][42], you create a username and password for a `default` namespace.
+During the [Sensu backend installation][42] process, you can set administrator credentials to override the default credentials for a `default` namespace.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/rbac.md]:
```suggestion
During the [Sensu backend installation][42] process, you can set administrator credentials to override the default credentials and a `default` namespace.
```

File: content/sensu-go/5.18/installation/install-sensu.md [hunk 7]
```diff
@@ -144,44 +159,18 @@ For a complete list of configuration options, see the [backend reference][6].
 
 ### 3. Initialize
 
+{{% notice note %}}
+**NOTE**: If you are using Docker, you already completed intitialization in [2. Configure and start](#2-configure-and-start).
+Skip ahead to [4. Open the web UI](#4-open-the-web-ui) to continue the backend installation process.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/installation/install-sensu.md:164]: WDYT about using this step to admonish folks to change their password if they used Docker defaults?

**Review by cwjohnston — CHANGES_REQUESTED** [review]
Default username and password details apply only to Docker deployments. Here are some suggestions to make that distinction clear.

File: content/sensu-go/5.18/installation/install-sensu.md (hunk 7 above)
Comment (hillaryfraley) [reply content/sensu-go/5.18/installation/install-sensu.md:164]: Yes, that's a good idea. I'll update to add it.

File: content/sensu-go/5.18/installation/install-sensu.md [hunk 8]
```diff
@@ -280,10 +273,12 @@ sensuctl configure -n \
 Here, the `-n` flag triggers non-interactive mode.
 Run `sensuctl config view` to see your user profile.
 
-We recommend that you change the default admin password immediately: `sensuctl user change-password --interactive`.
-
 For more information about sensuctl, see the [quickstart][23] and [reference][4] docs.
 
+### Change default admin password
+
+If you are using Docker and you did not use environment variables to override the default admin credentials in [step 2 of the backend installation process](#2-configure-and-start), we recommend that you change the default admin password immediately: `sensuctl user change-password --interactive`.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/installation/install-sensu.md]: Will they have completed initial sensuctl configuration at this point? I don't think they can change password without having run `sensuctl configure`

File: content/sensu-go/5.18/reference/rbac.md [hunk 9]
```diff
@@ -242,9 +242,9 @@ Use your Sensu username and password to [configure sensuctl][26] or log in to th
 
 ### Default users
 
-When you install the Sensu backend, during the [initialization step][42], you create a username and password for a `default` namespace.
+During the [Sensu backend installation][42] process, you can set administrator credentials to override the default credentials and create a `default` namespace.
```
Comment (cwjohnston) [inline content/sensu-go/5.18/reference/rbac.md]:
```suggestion
During the [Sensu backend installation][42] process, you create an administrator username and password and a `default` namespace.
```

**Review by cwjohnston — CHANGES_REQUESTED** [review]

File: content/sensu-go/5.18/installation/install-sensu.md (hunk 8 above)
Comment (hillaryfraley) [reply content/sensu-go/5.18/installation/install-sensu.md]: Yes, I need to be more specific about that. I will update the instruction.

**Review by cwjohnston — APPROVED** [review]

**Comment by hillaryfraley** [comment]
:tada:

**Pull request merged**